Client/Ecosia/Experiments/Unleash/EngagementServiceExperiment.swift

```diff
@@ -0,0 +1,30 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+import Foundation
+import Core
+
+struct EngagementServiceExperiment {
+    
+    private init() {}
+
+    static var isEnabled: Bool {
+        Unleash.isEnabled(.brazeEngagementService)
```

Comment:
I have to say `.brazeEngagementService` (`mob_ios_engagement_service_braze`) seems too vague for a flag name for me. Since this is about our apn consent UI and not necessarily even related to braze, I would prefer the flag to be something like `mob_ios_apn_consent_ui` in a descriptive way. What do you think?

Sorry I didn't mention it on the PR where the key was created since I reviewed it quite quickly to unblock you 🙈

Reply:
Thanks for the explanation! I understand the reasoning behind leaving the flag specific to a service, though I think that is very unlikely to become a problem, I have no issues in doing it to be safe 👍 

I would still say the flag needs to make it clear this is about the apn consent ui and would suggest then just adding `braze` to my previous suggestion, like `mob_ios_apn_consent_ui_braze`. What do you think?

Reply:
The UI part is an interesting point.
The way the flag works now limits the availability of the entire service, not just the UI.
I believe the role of the flag won't be solely delegated to the UI part as we wouldn't want to initialize Braze (as always required during app startup and not only at the consent step) if we plan to show the consent screen on the dedicated conditions only (e.g. non-core markets).
Do you think adding UI would explicitly help clarify and acknowledge that if set to `false` the end User won't have the service initialized either? I'm questioning that thinking product-wise and whoever will interact/read those flags on Unleash.
Thanks 🙏

Reply:
As far as I understood, we are utilizing Braze for Push Notification system only, so, no need today to have it initialized (as it will also create a binding with the Analytics ID).
I have to admit though that a clarification is needed. In case of utilizing the flag for the popup only, I'm absolutely onboard with you for a naming update on the Core side and reflected here (would inffact have chosen a similar name in the first place as well 😅 👍).

Reply:
@lucaschifino will make a dedicated, smaller PR for the toggle name change.